core/trino-main/src/main/java/io/trino/execution/resourcegroups/IndexedPriorityQueue.java

```diff
@@ -46,6 +46,9 @@ public boolean addOrUpdate(E element, long priority)
     {
         Entry<E> entry = index.get(element);
         if (entry != null) {
+            if (entry.getPriority() == priority) {
```

Comment:
this shouldn't be needed

Reply:
WDYM? It doesn't mean that if elements are equal then priorities are also equal.

Reply:
I mean:
```
if (entry.getPriority() == priority) {
  return false;
}
```
is really equivalent to:
```
            queue.remove(entry);
            Entry<E> newEntry = new Entry<>(element, priority, entry.getGeneration());
            queue.add(newEntry);
            index.put(element, newEntry);
            return false;
```
because `entry.getValue` and `element` are equal.
What you did here is an optimization, which is fine. However, it's was actually also required for your code to work correctly (I think it's not required anymore for code correctness)

Reply:
> What you did here is an optimization, which is fine. However, it's was actually also required for your code to work correctly (I think it's not required anymore for code correctness in newest revision)

Yeah, it's just an optimization since we could be doing a lot of `addOrUpdate` with the same priority in case of minSkewedWriter. However, it doesn't affect the implementation.